sql/hive/v0.12.0/src/main/scala/org/apache/spark/sql/hive/Shim12.scala

```diff
@@ -135,7 +135,13 @@ private[hive] object HiveShim {
       PrimitiveCategory.VOID, null)
 
   def getStringWritable(value: Any): hadoopIo.Text =
-    if (value == null) null else new hadoopIo.Text(value.asInstanceOf[UTF8String].toString)
+    if (value == null) {
+      null
+    } else if (value.isInstanceOf[String]) {
+      new hadoopIo.Text(value.asInstanceOf[String])
+    } else {
+      new hadoopIo.Text(value.asInstanceOf[UTF8String].toString)
+    }
```

Comment:
if want to return the Literal(database, StringType) in HiveFunctionRegistry.lookupFunctio, i need add a judge for current_databse. And this alse seems a hack way. any idea? @chenghao-intel